js/piwik.js

```diff
@@ -7208,16 +7210,16 @@ if (typeof window.Piwik !== 'object') {
              */
             this.requireConsent = function () {
                 if (!this.hasRememberedConsent()) {
-                    configRequireConsent = true;
+                    configHasConsent = false;
                 }
                 // Piwik.addPlugin might not be defined at this point, we add the plugin directly also to make JSLint happy
                 // We also want to make sure to define an unload listener for each tracker, not only one tracker.
                 coreConsentCounter++;
                 plugins['CoreConsent' + coreConsentCounter] = {
                     unload: function () {
-                        if (configRequireConsent) {
+                        if (!configHasConsent) {
                             // we want to make sure to remove all previously set cookies again
-                            deleteCookies();
+                            deleteCookies(['consent_removed']);
```

Comment:
Maybe we also ignore `consent`? In theory it should probably not make a difference, but may be still good to have?